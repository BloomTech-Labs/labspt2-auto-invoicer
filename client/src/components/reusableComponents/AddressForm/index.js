import React, { Component } from "react";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";

class index extends Component {
  constructor() {
    super();
    this.state = {
      invoiceFrom: "",
      invoiceTo: "",
      address1: "",
      city: "",
      state: "",
      zip: "",
      country: ""
    };
  }

  changeHandler = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  onSubmit = event => {
    event.preventDefault();
    this.props.onSubmit(this.state);
  };

  render() {
    return (
      <div>
        <Grid container spacing={16}>
          <Grid item xs={12} sm={6}>
            <TextField
              id="invoiceFrom"
              name="invoiceFrom"
              label="Invoice FROM"
              value={this.state.firstName}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12} sm={6}>
            <TextField
              id="invoiceTo"
              name="invoiceTo"
              label="Invoice TO"
              value={this.state.lastName}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              id="address1"
              name="address1"
              label="Address line 1"
              fullWidth
              value={this.state.address1}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12} sm={6}>
            <TextField
              id="city"
              name="city"
              label="City"
              autoComplete="billing address-level2"
              value={this.state.city}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12} sm={6}>
            <TextField
              id="state"
              name="state"
              label="State/Province/Region"
              value={this.state.state}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12} sm={6}>
            <TextField
              id="zip"
              name="zip"
              label="Zip / Postal code"
              autoComplete="billing postal-code"
              value={this.state.zip}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
          <Grid item xs={12} sm={6}>
            <TextField
              id="country"
              name="country"
              label="Country"
              autoComplete="billing country"
              value={this.state.country}
              onChange={event => this.changeHandler(event)}
            />
          </Grid>
        </Grid>
        <button onClick={event => this.onSubmit(event)}>Submit</button>
      </div>
    );
  }
}

export default index;
