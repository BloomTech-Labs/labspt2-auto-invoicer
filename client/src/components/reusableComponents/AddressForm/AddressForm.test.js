import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import AddressForm from "../AddressForm";

describe("<AddressForm />", () => {
  it("renders Address Form", () => {
    const { asFragment } = render(<AddressForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
