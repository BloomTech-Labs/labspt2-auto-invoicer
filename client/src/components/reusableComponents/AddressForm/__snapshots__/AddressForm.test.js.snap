// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressForm /> renders Address Form 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="invoiceFrom"
          >
            Invoice FROM
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="invoiceFrom"
              name="invoiceFrom"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="invoiceTo"
          >
            Invoice TO
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="invoiceTo"
              name="invoiceTo"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <div
          class="MuiFormControl-root-98 MuiFormControl-fullWidth-101"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="address1"
          >
            Address line 1
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-fullWidth-142 MuiInput-fullWidth-127 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="address1"
              name="address1"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="city"
          >
            City
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              autocomplete="billing address-level2"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="city"
              name="city"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="state"
          >
            State/Province/Region
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="state"
              name="state"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="zip"
          >
            Zip / Postal code
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              autocomplete="billing postal-code"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="zip"
              name="zip"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49"
      >
        <div
          class="MuiFormControl-root-98"
        >
          <label
            class="MuiFormLabel-root-113 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110"
            data-shrink="false"
            for="country"
          >
            Country
          </label>
          <div
            class="MuiInputBase-root-133 MuiInput-root-120 MuiInput-underline-124 MuiInputBase-formControl-134 MuiInput-formControl-121"
          >
            <input
              aria-invalid="false"
              autocomplete="billing country"
              class="MuiInputBase-input-143 MuiInput-input-128"
              id="country"
              name="country"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <button>
      Submit
    </button>
  </div>
</DocumentFragment>
`;
